lib/src/autosnapshotting/model.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:path/path.dart' as path;
+
+/// A record of a taken snapshot.
+class SnapshotRecord {
+  SnapshotRecord(
+    this.fileName, {
+    required this.snapshotNumber,
+    required this.rss,
+    DateTime? timestamp,
+  }) : timestamp = timestamp ?? DateTime.now();
+
+  final String fileName;
+  final int snapshotNumber;
+  final int rss;
+  final DateTime timestamp;
+}
+
+/// A callback that is called when a snapshot is taken.
+typedef SnapshotCallback = void Function(SnapshotRecord record);
+
+/// Configures auto-snapshotting, based on the value of ProcessInfo.currentRss (dart:io).
+///
+/// The snapshots will be taken as soon as the value becomes more than [thresholdMb],
```

Comment:
docs nit for clarity: "Automatic snapshots will begin to be taken when the rss value is exceeds [threshholdMb]. Snapshots will be saved to [directory]."

Can you also add more explanation for [directory]? What is this path relative to?